backpack/extensions/firstorder/batch_grad/batch_grad_base.py

```diff
@@ -15,26 +16,48 @@ class BatchGradBase(FirstOrderModuleExtension):
     In this case, the method is not overwritten by this class.
     """
 
-    def __init__(self, derivatives, params=None):
+    def __init__(self, derivatives, params):
         """Initializes all methods.
 
         If the param method has already been defined, it is left unchanged.
 
         Args:
             derivatives(backpack.core.derivatives.basederivatives.BaseParameterDerivatives):# noqa
                 Derivatives object assigned to self.derivatives.
-            params (list[str]): list of strings with parameter names. Defaults to None.
+            params (list[str]): list of strings with parameter names.
+                For each, a method is assigned.
         """
         self.derivatives = derivatives
         for param_str in params:
             if not hasattr(self, param_str):
-                self.__setattr__(param_str, self._make_function(param_str))
+                self.__setattr__(param_str, self._make_param_function(param_str))
         super().__init__(params=params)
 
-    def _make_function(self, param):
-        def function(ext, module, g_inp, g_out, bpQuantities):
+    def _make_param_function(self, param):
+        """Creates a function that calculates batch_grad wrt param.
+
+        Args:
+            param(str): name of parameter
+
+        Returns:
+            function: function that calculates batch_grad wrt param
+        """
+
+        def param_function(ext, module, g_inp, g_out, bpQuantities):
+            """Calculates batch_grad with the help of derivatives object.
+
+            Args:
+                ext(backpack.extensions.BatchGrad): extension that is used
+                module(torch.nn.Module): module that performed forward pass
+                g_inp(tuple[torch.Tensor]): input gradient tensors
+                g_out(tuple[torch.Tensor]): output gradient tensors
+                bpQuantities(None): additional quantities for second order
+
+            Returns:
+                torch.Tensor: individual gradients
```

Comment:
scaled individual gradients, as there might be a factor `1 / batch_size`, depending on the loss reduction